src/libs/actions/PersistedRequests.js

```diff
@@ -17,7 +19,12 @@ function clear() {
  * @param {Array} requestsToPersist
  */
 function save(requestsToPersist) {
-    persistedRequests = persistedRequests.concat(requestsToPersist);
+    HttpUtils.cancelPendingReconnectAppRequest();
+    persistedRequests = _.chain(persistedRequests)
+        .concat(requestsToPersist)
+        .partition(request => request.command !== CONST.NETWORK.COMMAND.RECONNECT_APP)
+        .flatten()
+        .value();
```

Comment:
What exactly is happening here? 🤔 

Is the goal to move the `ReconnectApp` calls to a different order? Why are we doing it this?

Reply:
You may need to take a look at the PR details or my linked proposal here https://github.com/Expensify/App/issues/15627 to get the context.

Yes, the goal is to re-order `ReconnectApp` to always fire last.

Whenever we switch from offline to online, `ReconnectApp` request is fired to retrieve the latest updates. Prior to this PR, `ReconnectApp` is fired first regardless of whether there's any request that need updating first:
1. The update action is made, the component is updated with optimistic data
2. `ReconnectApp` retrieves the old data (data before the update), the component is updated as the old data
3. Update request retrieves the updated data, the component is updated as the success data

This causes the component to flicker between optimistic, old and success data. Note that if we do this, the order of other requests will remain the same.

For example:

```
A > B > ReconnectApp > C
```
to
```
A > B > C > ReconnectApp
```